linera-chain/src/certificate.rs

```diff
@@ -69,13 +75,25 @@ impl<T: Debug> Debug for GenericCertificate<T> {
     }
 }
 
-// NOTE: For backwards compatiblity reasons we serialize the new `Certificate` type as the old
-// one. Otherwise we would be breaking the RPC API schemas. We can't implement generic serialization
-// for `Certificate<T>` since only specific `T`s have corresponding `CertificateValue` variants.
 impl Serialize for ValidatedBlockCertificate {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let cert = Certificate::from(self.clone());
-        cert.serialize(serializer)
+        let mut state = serializer.serialize_struct("ValidatedBlockCertificate", 4)?;
+        state.serialize_field("hash", &self.value.hash())?;
+        state.serialize_field("value", &self.inner())?;
+        state.serialize_field("round", &self.round)?;
+        state.serialize_field("signatures", &self.signatures)?;
+        state.end()
+    }
+}
+
+impl Serialize for TimeoutCertificate {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut state = serializer.serialize_struct("TimeoutCertificate", 4)?;
+        state.serialize_field("hash", &self.value.hash())?;
```

Comment:
Wait, why are we serializing the hash? We must not ever trust a deserialized hash we received from an untrusted source (i.e. anything other than a machine in the same validator cluster).

Reply:
Yes, you're right of course. I kept this approach from previous PR - since there I was still mapping to/from old `Certificate` types but here I don't do that anymore. I'll update the code so that hash is not serialized and upon deserialization we verify it.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2845/commits/6d6d8828a92f7c32a8d9e7fc8e4bfaf8b9eba13a